variety/__init__.py

```diff
@@ -25,16 +25,16 @@
 def _u(s):
     if s is None:
         return s
-    if isinstance(s, unicode):
+    if isinstance(s, str):
```

Comment:
These two functions (_u converts to a string, _str - to a bytes array) will need changing (and potentially renamed as _u and _bytes), and all of their usages need to be checked. Basically they get around the Unicode deficiencies of Python 2 (and of my own code in places it wasn't thought in time for unicode vs. str aspects). With Python 3 we may want to use them in a lot less places and be more explicit about whether something is a string or a byte array.